aten/src/ATen/native/ComplexHelper.h

```diff
@@ -48,8 +48,9 @@ Tensor view_as_complex(const Tensor& self) {
     self.scalar_type() == kFloat || self.scalar_type() == kDouble || self.scalar_type() == kHalf,
     "view_as_complex is only supported for half, float and double tensors, but got a tensor of scalar type: ", self.scalar_type());
 
+  TORCH_CHECK(self.dim() != 0, "Input tensor must have one or more dimensions");
   auto new_sizes = self.sizes().vec();
-  TORCH_CHECK(new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");
+  TORCH_CHECK(self.dim() != 0 && new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");
```

Comment:
Redundant check for non zero dimension can be avoided, since it has already been checked above.

Reply:
@guol-fnst yeah we should remove the check here

Reply:
Yes, this check has benn removed